Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx

```diff
@@ -40,31 +43,71 @@ struct SelectTracks {
+  Configurable<LabeledArray<double>> cutsTrack3Prong{"cuts_singletrack_3prong", {hf_cuts_single_track::cutsTrack[0], npTBinsTrack, nCutVarsTrack, pTBinLabelsTrack, cutVarLabelsTrack}, "Single-track selections per pT bin for 3-prong candidates"};
   Configurable<double> etamax_3prong{"etamax_3prong", 4., "max. pseudorapidity for 3 prong candidate"};
 
   HistogramRegistry registry{
     "registry",
     {{"hpt_nocuts", "all tracks;#it{p}_{T}^{track} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
      // 2-prong histograms
      {"hpt_cuts_2prong", "tracks selected for 2-prong vertexing;#it{p}_{T}^{track} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
-     {"hdcatoprimxy_cuts_2prong", "tracks selected for 2-prong vertexing;DCAxy to prim. vtx. (cm);entries", {HistType::kTH1F, {{static_cast<int>(1.2 * dcatoprimxymax_2prong * 100), -1.2 * dcatoprimxymax_2prong, 1.2 * dcatoprimxymax_2prong}}}},
+     {"hdcatoprimxy_cuts_2prong", "tracks selected for 2-prong vertexing;DCAxy to prim. vtx. (cm);entries", {HistType::kTH1F, {{400, -2., 2.}}}},
      {"heta_cuts_2prong", "tracks selected for 2-prong vertexing;#it{#eta};entries", {HistType::kTH1F, {{static_cast<int>(1.2 * etamax_2prong * 100), -1.2 * etamax_2prong, 1.2 * etamax_2prong}}}},
      // 3-prong histograms
      {"hpt_cuts_3prong", "tracks selected for 3-prong vertexing;#it{p}_{T}^{track} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}}},
-     {"hdcatoprimxy_cuts_3prong", "tracks selected for 3-prong vertexing;DCAxy to prim. vtx. (cm);entries", {HistType::kTH1F, {{static_cast<int>(1.2 * dcatoprimxymax_3prong) * 100, -1.2 * dcatoprimxymax_3prong, 1.2 * dcatoprimxymax_3prong}}}},
+     {"hdcatoprimxy_cuts_3prong", "tracks selected for 3-prong vertexing;DCAxy to prim. vtx. (cm);entries", {HistType::kTH1F, {{400, -2., 2.}}}},
      {"heta_cuts_3prong", "tracks selected for 3-prong vertexing;#it{#eta};entries", {HistType::kTH1F, {{static_cast<int>(1.2 * etamax_3prong * 100), -1.2 * etamax_3prong, 1.2 * etamax_3prong}}}}}};
 
+  /// Single-track cuts for 2-prongs
+  /// \param hfTrack is a track
+  /// \return true if track passes all cuts
+  template <typename T>
+  bool selectionTrack2Prong(const T& hfTrack, const array<float, 2>& dca)
+  {
+    auto trackpT = hfTrack.pt();
+    auto pTBinTrack = findBin(pTBinsTrack, trackpT);
+    if (pTBinTrack == -1) {
+      return false;
+    }
+
+    if (dca[0] < cutsTrack2Prong->get(pTBinTrack, "min_dcaxytoprimary")) {
+      return false; //minimum DCAxy
+    }
+    if (dca[0] > cutsTrack2Prong->get(pTBinTrack, "max_dcaxytoprimary")) {
+      return false; //maximum DCAxy
+    }
+    return true;
+  }
+
+  /// Single-track cuts for 3-prongs
+  /// \param hfTrack is a track
+  /// \return true if track passes all cuts
+  template <typename T>
+  bool selectionTrack3Prong(const T& hfTrack, const array<float, 2>& dca)
```

Comment:
The only advantage in having it implemented as it is now is that we spare two `if` statements per track to the price of a bit of code duplication, even if the additional time for them is likely negligible. Let me know if you prefer to go in the direction of remove the code duplication and have the additional `if` statements I can implement it accordingly

Reply:
I don't think any additional `if` statements are needed at all. The switch can be a template argument and the corresponding cut array can be picked inside the function by index. Maybe there are better ways, but that was the first thing that came to my mind on how to avoid duplication and conditions at the same time.